server/chat-jsx.tsx

```diff
@@ -36,3 +37,35 @@ export type VNode = preact.VNode;
 
 export const h = preact.h;
 export const Fragment = preact.Fragment;
+
+function subRender(
+	child: preact.ComponentChild, opts?: {isTrusted?: boolean, replaceLinebreaks?: boolean}
+): preact.VNode {
+	switch (typeof child) {
+	case 'string':
+		return <div dangerouslySetInnerHTML={{__html: Chat.formatText(child, opts?.isTrusted, opts?.replaceLinebreaks)}} />;
+	case 'bigint':
+	case 'number':
+	case 'boolean':
+	case 'object':
+		return <>{child ? child.toString() : '' + child}</>;
+	default:
+		return preact.isValidElement(child) ? child : <>{'' + child}</>;
+	}
+}
+
+export class FormatText extends preact.Component<{isTrusted?: boolean, replaceLinebreaks?: boolean}> {
+	render() {
+		if (Array.isArray(this.props.children)) {
```

Comment:
I don't see a need to handle all these special cases. Just throw if `children` isn't a string.

Reply:
I guess that works.

Reply:
This does not seem resolved.

Reply:
I am not throwing on arrays because components can take arrays as children

Reply:
Yes, but it's a bug if they do. That's why this should throw. The purpose of throwing is to tell you that your code has a bug.

Reply:
I do think taking the main text in a `text` attribute rather than in `children` would be better, if this is a problem.

Reply:
That's fair, will change. I don't think we need to use a text attribute for now, but that could change.